Doc/persistence

jannyHou opened the pull request:
connect to strongloop-internal/scrum-apex#229

Break down the task into small PRs due to frequent underneath code check in(conflicts)
- [x] repository.ts
- [x] query.ts

ssh24 commented on packages/repository/src/repository.ts:
```diff
@@ -113,24 +124,48 @@ export interface EntityCrudRepository<T extends Entity, ID>
    * Check if an entity exists for the given id
+   * @param id
+   * @param options
+   * @promise {Boolean} resolves a Boolean
```
Shouldn't all the params have the type of the param on the docs specified with a small description? Something like this:

```
@param id {ID} The `id` value
```

jannyHou replied:
@ssh24 I am confirming with the jsdoc annotation standard we follow.

jannyHou commented:
I am not sure why it complains PR out of date
After I git pull feature/persistence and rebase my branch, everything is update-to-date:

```shell
remote: Counting objects: 38, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 38 (delta 16), reused 22 (delta 16), pack-reused 14
Unpacking objects: 100% (38/38), done.
From https://github.com/strongloop/loopback-next
   f270b91..b846642  injectable-sequence -> origin/injectable-sequence
   0ae748e..67841d3  master              -> origin/master
 + cf71f62...680130f providerInterface   -> origin/providerInterface  (forced update)
Already up-to-date.
bash-3.2$ git checkout doc/persistence
Switched to branch 'doc/persistence'
Your branch is up-to-date with 'origin/doc/persistence'.
bash-3.2$ git rebase -i feature/persistence
Successfully rebased and updated refs/heads/doc/persistence.
bash-3.2$ git push -f
Everything up-to-date
```

jannyHou commented:
I am closing this one
